chsdi/templates/htmlpopup/cadastralwebmap.mako

```diff
@@ -78,4 +88,174 @@
     % else:
         <tr><td class="cell-left">${_('link to canton geoportal')}</td><td>${_('Canton has provided no link to portal')}</td></tr>
     % endif
+       <tr>
+       <td class="cell-left">${_('ch.kantone.cadastralwebmap-farbe.plan')}</td>
+       <td style="height: 20px;">
+         <iframe src="${baseUrl}/rest/services/all/MapServer/${c['layerBodId']}/${c['featureId']}/extendedHtmlPopup?lang=${lang}&geometry=${request.params.get('coord')}&iframe=true" width="100%" height="20" frameborder="0" style="border: 0; position:absolute; z-index:1" scrolling="no" onload="$(this).next().hide();"></iframe>
+         <div style="position: absolute; z-index:2;">
+           <p>${_('ch.kantone.cadastralwebmap-farbe.wait')}</p>
+         </div>
+       </td>
+       </tr>
+
+       <tr>
+          <td class="cell-left"></td>
+          <td>
+            <a href="${h.make_agnostic(request.route_url('extendedHtmlPopup', map=topic, layerId=c['layerBodId'], featureId=str(c['featureId'])))}?lang=${lang}&geometry=${request.params.get('coord')}" target="_blank">${_('zusatzinfo')}&nbsp;<img src="${h.versioned(request.static_url('chsdi:static/images/ico_extern.gif'))}" /></a>
+          </td>
+        </tr>
+</%def>
+
+<%def name="extended_info(c, lang)">
+<%
+    import requests
+    import time
+    
+    from chsdi.models.vector import getScale
+    from chsdi.lib.validation.features import IdentifyServiceValidation
+    request = context.get('request')
+    defaultExtent = '42000,30000,350000,900000'
+    defaultImageDisplay = '400,600,96'
+    fallbackLang = 'fr' if request.lang in ('fr', 'it') else 'de'
+    class CadastralWebMapParams(IdentifyServiceValidation):
+        def __init__(self, request):
+            self.mapExtent = request.params.get('mapExtent', defaultExtent)
+            self.imageDisplay = request.params.get('imageDisplay', defaultImageDisplay)
+    params = CadastralWebMapParams(request)
```

Comment:
This is a lot of duplicated code. You can use:
<%! %>
to make the common parts of the code available to the whole template.